napari/layers/surface/wireframe.py

```diff
@@ -0,0 +1,25 @@
+from typing import Union
+
+from ...utils.events import EventedModel
+from ...utils.events.custom_types import Array
+
+
+class SurfaceWireframe(EventedModel):
+    """
+    Wireframe representation of the edges of a surface mesh.
+
+    Attributes
+    ----------
+    visible : bool
+        Whether the wireframe is displayed.
+    color : str, array-like
+        If string can be any color name recognized by vispy or hex value if
+        starting with `#`. If array-like must be 1-dimensional array with 3
+        or 4 elements.
+    width : float
+        The width of the wireframe lines.
+    """
+
+    visible: bool = False
+    color: Union[str, Array[float, (3,), Array[float, (4,)]]] = 'black'
```

Comment:
really wish we could somehow standardize this :/
we accept colors _so_ many places... it would be great if we didn't still have to define them as unions of various valid inputs.  (among other things, this means that, as far as the typing implies, the _consumer_ of `wireframe.color` needs to be able to handle any of these options).

if you can't use `pydantic.color.Color`, maybe you can start to think about a napari color field?  @andy-sweet has also looked into this and knows a lot about the sneaky pitfalls.  but just pinning it again.

Reply:
I usually think about the declared type as the one that will be stored. For a single color value in napari, I think that's usually `Array[float, (4,)]` (and is almost the type returned by `transform_color` - that actually has shape (-1, 4)).

For supported parameter/inputs, we have `ColorType`, but that includes vispy's `Color`, which isn't currently supported by `transform_color` (though could be easily).

I started writing a developer guide around colors in napari, which might be helpful to understand some of the details here: https://hackmd.io/K5rCILaHQmWsNhY0ml_xaQ

In the style encoding work, I defined a custom types for a single `ColorValue` and multiple `ColorArray`: https://github.com/napari/napari/pull/3493/files#diff-c6c515811f47fb2a55c3edb7f778cc93b0bbf1435d21c11a11e78a3d6d1e7ba5R23

I'd be nervous about bringing in pydantic's `Color` at this point (even though it's great!), because it would add some inconsistency (e.g. it supports CSS3 color names, but it doesn't support matplotlib's single character names).

Reply:
I agree that this is not good; is this something you consider blocking for this PR and we should work on first?

Reply:
I don't think this feature should be blocked until colours are standardised, it would presumably be a fairly easy thing to update once progress was made on colours?

Reply:
Yeah, definitely not blocking